feat: support native type stripping for Node >= v23 (#442)
fastify/fastify-autoload: README.md

  ```

## Override TypeScript detection using an environment variable

> > This plugin uses [native type stripping](https://nodejs.org/docs/latest-v23.x/api/typescript.html#modules-typescript) with Node 23 and later.

Reviewer: ```suggestion
This plugin uses [native type stripping](https://nodejs.org/docs/latest-v23.x/api/typescript.html#modules-typescript) with Node 23 and later.
```

Don't think this needs the block quote to draw attention to it. It's the first sentence of a section, redaers are going to see it.

Author: This is not about drawing attention, just a note that is not part of the main explanations. But ok.

> > **Note**: Node >= 23 supports native type stripping, and this plugin will automatically take advantage of it. Read more about native type stripping on the [API docs](https://nodejs.org/docs/latest-v23.x/api/typescript.html#modules-typescript).

Reviewer: ```suggestion

This plugin uses [native type stripping](https://nodejs.org/docs/latest-v23.x/api/typescript.html#modules-typescript) with Node 23 and later.
```
